lib/elements/reinforcement/straight_longitudinal.rb

```diff
@@ -0,0 +1,101 @@
+require 'elements/reinforcement/base'
+require 'elements/reinforcement/straight_longitudinal_layer'
+require 'errors/reinforcement/empty_layers'
+
+module Elements
+  module Reinforcement
+    class StraightLongitudinal < Base
+      def initialize(distribution_direction:, above_middle: false)
+        @above_middle = above_middle
+        @layers = []
+        @distribution_direction = distribution_direction
+      end
+
+      def add_layer(start_location:, end_location:, amount_of_rebars:, rebar:)
+        id = @layers.empty? ? 1 : @layers.last.id
+
+        start_location = modify_value_3_of_location(start_location, 0.5 * rebar.diameter)
+        end_location = modify_value_3_of_location(end_location, 0.5 * rebar.diameter)
+
+        @layers << Elements::Reinforcement::StraightLongitudinalLayer.new(
+          start_location:,
+          end_location:,
+          amount_of_rebars:,
+          rebar:,
+          id:,
+          distribution_direction: @distribution_direction
+        )
+
+        @layers.last
+      end
+
+      def change_layer_rebar_configuration(
+        id_of_layer_to_change:,
+        amount_of_new_rebars:,
+        new_rebar:
+      )
+        straight_longitudinal_layer = find(id_of_layer_to_change)
+
+        offset = (straight_longitudinal_layer.diameter - new_rebar.diameter) / 2
+        offset *= -1 unless @above_middle
+
+        straight_longitudinal_layer.modify_rebar_configuration(
+          amount_of_new_rebars:,
+          new_rebar:,
+          offset:
+        )
+
+        straight_longitudinal_layer
+      end
+
+      def move_layer_by_its_axis_3(id_of_layer_to_change:, offset:)
+        offset *= -1 if @above_middle
+
+        straight_longitudinal_layer = find(id_of_layer_to_change)
+        straight_longitudinal_layer.move_axis_3(offset:)
+
+        straight_longitudinal_layer
+      end
+
+      def centroid_height
+        if @layers.empty?
+          raise Elements::Reinforcement::EmptyLayers, "can't complete centroid height calculation"
+        end
+
+        inertia = 0
+        total_area = 0
+
+        @layers.each do |layer|
+          total_area += layer.area
+          inertia += layer.inertia
+        end
+
+        inertia / total_area
```

Comment:
I see you have an `area` method defined below. Isn't it the same as the `total_area` you are trying to calculate here? If so, I think you can also define a `inertia` method that is the sum of all inertias and by using both methods this one would be pretty straightforward

Reply:
But, in that way, we will go twice through the @layers array, no?

Reply:
I think it's fine to sacrifice performance for better readability. The `@layers` array is not really going to be huge

Reply:
agree, done!